src/amo/components/AddonReviewListItem/index.js

```diff
@@ -1,55 +1,125 @@
 import { connect } from 'react-redux';
 import { compose } from 'redux';
 
 import AddonReview from 'amo/components/AddonReview';
+import { withErrorHandler } from 'core/errorHandler';
 import translate from 'core/i18n/translate';
+import log from 'core/logger';
 import { isAuthenticated } from 'core/reducers/user';
-import { nl2br, sanitizeHTML } from 'core/utils';
+import { isAddonAuthor, nl2br, sanitizeHTML } from 'core/utils';
+import {
+  hideEditReviewForm,
+  hideReplyToReviewForm,
+  sendReplyToReview,
+  showEditReviewForm,
+  showReplyToReviewForm,
+} from 'amo/actions/reviews';
 import LoadingText from 'ui/components/LoadingText';
 import Rating from 'ui/components/Rating';
+import DismissibleTextForm from 'ui/components/DismissibleTextForm';
 import type { UserReviewType } from 'amo/actions/reviews';
+import type { ReviewState } from 'amo/reducers/reviews';
+import type { ErrorHandlerType } from 'core/errorHandler';
 import type { UserStateType } from 'core/reducers/user';
+import type { AddonType } from 'core/types/addons';
+import type { DispatchFunc } from 'core/types/redux';
+import type { OnSubmitParams } from 'ui/components/DismissibleTextForm';
 
 import './styles.scss';
 
 type PropsType = {|
+  addon?: AddonType,
+  editingReview: boolean,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
   isAuthenticated: boolean,
   i18n: Object,
-  review: UserReviewType,
+  review?: UserReviewType,
+  replyingToReview: boolean,
   siteUser: UserStateType,
+  submittingReply: boolean,
 |};
 
 export class AddonReviewListItemBase extends React.Component {
   props: PropsType;
-  state: {| editingReview: boolean |};
-
-  constructor(props: PropsType) {
-    super(props);
-    this.state = { editingReview: false };
-  }
 
   onClickToEditReview = (event: SyntheticEvent) => {
+    const { dispatch, review } = this.props;
     event.preventDefault();
-    this.setState({ editingReview: true });
+    if (!review) {
+      log.debug('Cannot edit a review because no review has been loaded.');
+      return;
+    }
+    dispatch(showEditReviewForm({ reviewId: review.id }));
   }
 
   onEscapeReviewOverlay = () => {
+    const { dispatch, review } = this.props;
+    if (!review) {
+      log.debug('Cannot hide review form because no review has been loaded.');
+      return;
+    }
     // Even though an escaped overlay will be hidden, we still have to
     // synchronize our show/hide state otherwise we won't be able to
     // show the overlay after it has been escaped.
-    this.setState({ editingReview: false });
+    dispatch(hideEditReviewForm({ reviewId: review.id }));
   }
 
   onReviewSubmitted = () => {
-    this.setState({ editingReview: false });
+    const { dispatch, review } = this.props;
+    if (!review) {
+      log.debug('Cannot hide review form because no review has been loaded.');
+      return;
+    }
+    dispatch(hideEditReviewForm({ reviewId: review.id }));
+  }
+
+  onClickToBeginReviewReply = () => {
```

Comment:
This (and other events here) should accept the `event` param and run `event.preventDefault()`. Otherwise `#reply` gets added to the URL which isn't needed:

![2017-10-04 15 23 39](https://user-images.githubusercontent.com/90871/31180996-144f421e-a918-11e7-949c-c5852e3d11ba.gif)